mesa/Person.py

```diff
@@ -333,15 +388,114 @@ def remove_from_household(self):
                 member.neighbors.get("household").remove(self)
                 self.neighbors.get("household").remove(member)
 
+    def candidates_weight(self, agent):
+        """
+        This is what in the paper is called r - this is r R is then operationalised as the proportion
+        of OC members among the social relations of each individual (comprising family, friendship, school,
+        working and co-offending relations)
+        :return: float, the candidates weight
+        """
+        return -1 * (extra.social_proximity(self,
+                                            agent) * self.oc_embeddedness() * self.criminal_tendency) if agent.oc_member \
+            else (extra.social_proximity(self, agent) * self.criminal_tendency)
 
-class Prisoner(Person):
-    sentence_countdown = 0
+    def _agents_in_radius(self, context=network_names):
+        """
+        It finds the agents distant 1 in the specified networks, by default it finds it on all networks.
+        :param context: list, of strings, limit to networks name
+        :return: set, of Person objects
+        """
+        agents_in_radius = set()
+        for net in context:
+            if self.neighbors.get(net):
```

Comment:
Line 410 checks that the network is not empty, if so it would not be possible to iterate in the agents of that network. Personally I don't think it is necessary to manage the error here, but if you think it is necessary I can add a check that the networks passed to the context argument exist and if not raise an error.

Reply:
I thought line 410 does not check for empty, but null? This is safer:
```
        for net in context:
            print("here")
            for agent in self.neighbors.get(net):
                agents_in_radius.add(agent)
        return agents_in_radius
```
.. (but without the print)! It doesn't fail if the network is empty, and doesn't hide the problem if you pass it the 'professonal' network for example

Reply:
Done!